backend/src/helpers/students.js

```diff
@@ -8,3 +29,31 @@ exports.addStudent = async function (student) {
     return { err }
   }
 }
+
+exports.patchStudent = async function (id, patchStudent) {
+  // debug("test update notes")
+  try {
+    const response = await db.Student.query()
+    // .patch(patchstudent)
+    // .findById(id)
+      .patchAndFetchById(id, patchStudent)
+      .throwIfNotFound()
+    return response
+  } catch (err) {
+    return { err }
+  }
+}
+
+exports.getStatusByStudentId = async function (studentID) {
+  try {
+    // Note that there is no `await` here. This query does not get executed.
+    const studentSubQuery = db.Student.query().where('StudentID', studentID)
+
+    // This is the only executed query in this example.
+    const status = await db.Student.relatedQuery('status')
+      .for(studentSubQuery)
```

Comment:
I believe you can get what you need with a query like this:
```js
const status = await db.Student.relatedQuery('status').for(studentId)
```
This should work because `status` was defined in the `relationMappings`.

[Reference](https://vincit.github.io/objection.js/guide/query-examples.html#relation-find-queries)

Reply:
`const status = await db.Student.relatedQuery('status').for(studentId)`

Yes this works and I have cleaned up the code. Cool! Thank you Thomas!